app/templates/components/request/config-form.hbs

```diff
@@ -0,0 +1,144 @@
+        <form.field
+          data-test-request-configs-branch="true"
+          @classNames="ref-type-branch"
+          @placeholder="Select a branch"
+          @value={{this.branch}}
+          @showValidationStatusIcons={{false}}
+          @onChange={{action "change" "branch"}}
+          as |field|
+        >
+          <field.select
+            @search={{perform this.searchBranches}}
+            as |branch|
+          >
+            {{branch}}
+          </field.select>
+        </form.field>
+      {{/if}}
+    </div>
+
+    <div class="form-elem">
+      <form.field
+        data-test-request-configs-message="true"
+        @classNames="ref-type-sha"
+        @value={{this.message}}
+        @label="Message"
+        @onChange={{action "change" "message"}}
+        @placeholder="Overwrite the commit message (optional)"
+        as |field|
+      >
+        {{field.input}}
+      </form.field>
+    </div>
+    <div class="form-elem">
+      <form.field
+        data-test-request-configs-config="true"
+        @label="Custom Build Config"
+        as |field|
+      >
+        <div class="config-type {{this.configMode}}">
+          <span class="yaml">YAML</span>
+          <span class="javascript">JSON</span>
+        </div>
+
+        {{ivy-codemirror
+          value=this.config
+          valueUpdated=(action "change" "config")
+          options=(hash
+            mode=this.configMode
+            lineNumbers=true
+            tabSize=2
+            indentWithTabs=false
+            placeholder="Overwrite build config"
+          )
+        }}
+
+        <Request::Messages
+          @request={{this.request}}
+          @messages={{this.messages}}
+          @loading={{this.loading}}
+        />
+
+      </form.field>
+    </div>
+
+    <div class="form-elem">
+      <h4>
+        Merge mode
+      </h4>
+      {{#radio-button
+        value="deep_merge_append"
+        groupValue=mergeMode
+        changed=(action "change" "mergeMode")
+      }}
+        Deep Merge & Append
+      {{/radio-button}}
+      {{#radio-button
+        value="deep_merge_prepend"
+        groupValue=mergeMode
+        changed=(action "change" "mergeMode")
+      }}
+        Deep Merge & Prepend
+      {{/radio-button}}
+      {{#radio-button
+        value="deep_merge"
+        groupValue=mergeMode
+        changed=(action "change" "mergeMode")
+      }}
+        Deep Merge
+      {{/radio-button}}
+      {{#radio-button
+        value="merge"
+        groupValue=mergeMode
+        changed=(action "change" "mergeMode")
+      }}
+        Merge
+      {{/radio-button}}
+      {{#radio-button
+        value="replace"
+        groupValue=mergeMode
+        changed=(action "change" "mergeMode")
+      }}
+        Replace
+      {{/radio-button}}
+
+      <ExternalLinkTo class="hint" href="{{config-get 'urls.triggerBuildMergeModes'}}">
+        <SvgImage @name='icon-help' @class='icon-help' />
+      </ExternalLinkTo>
+    </div>
+  </div>
+</TravisForm>
```

Comment:
Is it by design that this form doesn't have a submit button? Or did I miss it?

Reply:
the button is included in the parent component `request/configs`, and was added in the previous branch. eventually there will be 4 different buttons, with one of them being the main submit `Trigger Build` button. this is how it looks like in the final branch: https://github.com/travis-ci/travis-web/blob/sf-request-config-tweaks/app/templates/components/request/configs-buttons.hbs#L26-L38

Reply:
I'm not sure, but does it mean that 
```hbs
<TravisForm
  @onSubmit={{perform this.triggerBuild}}
  as |form|
>
```
doesn't make sense as `onSubmit` will never be triggered?